src/bun.js/api/bun.zig

```diff
@@ -525,6 +527,54 @@ pub fn shell(
     // };
 }
 
+pub fn shellEscape(
+    globalThis: *JSC.JSGlobalObject,
+    callframe: *JSC.CallFrame,
+) callconv(.C) JSC.JSValue {
+    if (callframe.argumentsCount() < 0) {
+        globalThis.throw("shell escape expected at least 1 argument", .{});
+        return .undefined;
+    }
+
+    const jsval = callframe.argument(0);
+    const bunstr = jsval.toBunString(globalThis);
+    defer bunstr.deref();
+
+    var outbuf = std.ArrayList(u8).init(bun.default_allocator);
+    defer outbuf.deinit();
+    if (bunstr.isUTF8()) {
```

Comment:
this wil always be false.

`isUTF8` only works for ZigString.

you want `.toSlice(bun.default_allocator)` and then check if it is allocated.